catalyst_voices/packages/internal/catalyst_voices_shared/lib/src/range/range.dart

```diff
@@ -44,3 +44,52 @@ class Range<T extends num> extends Equatable {
   @override
   List<Object?> get props => [min, max];
 }
+
+class DateRange extends Equatable {
+  final DateTime? from;
+  final DateTime? to;
+
+  const DateRange({
+    required this.from,
+    required this.to,
+  });
+
+  bool isInRange(DateTime value) {
+    if (from == null && to != null) {
+      return value.isBefore(to!) || value.isAtSameMomentAs(to!);
+    }
+
+    if (from != null && to == null) {
+      return value.isAfter(from!) || value.isAtSameMomentAs(from!);
+    }
+
+    if (from != null && to != null) {
+      return (value.isAtSameMomentAs(from!) || value.isAfter(from!)) &&
+          (value.isAtSameMomentAs(to!) || value.isBefore(to!));
+    }
+
+    return true;
+  }
```

Comment:
Conceptually you need to check whether `from <= value && value <= to`, try to simplify it until you reach something similar.

```suggestion
  bool isInRange(DateTime value) {
    final min = from?.millisecondsSinceEpoch ?? 0;
    final max = to?.millisecondsSinceEpoch ?? DateTime(2099).millisecondsSinceEpoch;
    final valueMillis = value.millisecondsSinceEpoch;
    
    return min <= valueMillis && valueMillis <= max;
}
```